src/env/fake.rs

```diff
@@ -0,0 +1,145 @@
+use std::cell::RefCell;
+use std::ffi::OsString;
+use std::fmt;
+use std::sync::Arc;
+
+use crate::{error::Error, parse_args, run, Args};
+
+use super::Env;
+
+/// A command to run in a [`FakeEnv`]
+///
+/// This is used for testing the utilities, running the real code in a fake
+/// environment.
+#[derive(Clone)]
+pub struct FakeCmd {
+    /// The (sub)command to run, including `argv[0]`
+    pub cmd: Vec<OsString>,
+
+    /// The arguments for the commands
+    pub args: Vec<OsString>,
```

Comment:
The idea was that you could create a command and call it with different arguments. But since `args` now extends the arguments this can be undone I suppose.